README.rst

```diff
@@ -348,120 +356,114 @@ For example:
-In this file, you can specify all rocks to build
+Second, ``cartridge`` installs all dependencies specified in the rockspec file.
+(That file is located in the application directory created from template.)
+
+Alternatively, you can define the build logic in the rockspec in the form of ``cmake`` commands,
+`like we do <https://github.com/tarantool/cartridge/blob/master/cartridge-scm-1.rockspec#L26>`_
+in ``cartridge``.)
+
+If your application depends on closed-source rocks, or if your build contains
+rocks from a project added as a submodule, install all those
+dependencies **before** calling ``tarantoolctl rocks make``.
+You can do so using the file ``cartridge.pre-build`` in your application root
+(again, that file is located in the application directory created from template).
+In this file, you can specify all the rocks to build
 (e.g. ``tarantoolctl rocks make --chdir ./third_party/proj``).
-For details, see `special files <Special files_>`_.
+For details, see `build and packing files <Special files_>`_.
 
-As a result, in the application's ``.rocks`` directory you will get a fully built
-application that you can start locally from the application's directory.
+As a result, a fully built application will appear in the ``.rocks`` directory.
+You can start it locally from the application directory.
 
 .. _cartridge-cli-starting-stopping-an-application-locally:
 
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Starting/stopping an application locally
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-**********
+*********
 ``start``
-**********
+*********
 
-Now, after the application is `built <Building an application_>`_,
+After you've `built <Building an application_>`_ your application,
 you can run it locally:
 
 .. code-block:: bash
 
     cartridge start [INSTANCE_NAME...] [flags]
 
-where ``[INSTANCE_NAME...]`` means that several instances can be specified.
+where ``[INSTANCE_NAME...]`` stands for one or multiple instance names.
 
-If no ``INSTANCE_NAME`` is provided, all the instances from the
-Cartridge instances configuration file are taken as arguments (see the ``--cfg``
+If no ``INSTANCE_NAME`` is provided, all the instances in the
+Cartridge instances configuration file will be taken as arguments (see the ``--cfg``
 option below).
 
-We also need an application name (``APP_NAME``) to pass it to the instances while
-started and to define paths to the instance files (for example, ``<run-dir>/<APP_NAME>.<INSTANCE_NAME>.pid``).
-By default, the ``APP_NAME`` is taken from the application rockspec in the current
-directory, but also it can be defined explicitly via the ``--name`` option
-(see description below).
+.. // TODO add links on every "below"
+
+The application name, ``APP_NAME``, is passed to the instances during startup and
+used in instance file paths (for example, ``<run-dir>/<APP_NAME>.<INSTANCE_NAME>.pid``).
```

Comment:
```suggestion
The application name, ``APP_NAME``, is passed to the instances during startup and
used in instance file paths.
For example, ``<run-dir>/<APP_NAME>.<INSTANCE_NAME>.pid``.
```

Reply:
The last sentence looks truncated, I would reformulate:
```suggestion
The application name, ``APP_NAME``, is passed to the instances during startup and
used in instance file paths, like in the following example: ``<run-dir>/<APP_NAME>.<INSTANCE_NAME>.pid``.
```